engine/eConstr.ml

```diff
@@ -651,6 +654,10 @@ let substnl subst n c = of_constr (Vars.substnl (List.map to_constr subst) n (to
 let substl subst c = of_constr (Vars.substl (List.map to_constr subst) (to_constr c))
 let subst1 c r = of_constr (Vars.subst1 (to_constr c) (to_constr r))
 
+let substnl_decl subst n d = of_rel_decl (Vars.substnl_decl (List.map to_constr subst) n (to_rel_decl d))
```

Comment:
> I thought the whole idea behind EConstr was that functions exposing defined evars should not be used.

We may indeed eventually consider removing the `Unsafe` submodule of `EConstr`?? (Though it may be useful for optimizations - the same way as we may want to use `unsafe_array_get` of the `Array` library.)

I see otherwise a second aspect in `EConstr`: to replace a `nf_evars/whd_evar` model which inline evars whenever possible by a model where we internally keep the sharing induced by evars but don't observe it.